using-timescaledb/distributed-hypertables.md

```diff
@@ -48,32 +54,175 @@ CREATE TABLE conditions (
+push down and do not involve a lot of data nodes are unlikely to
+perform better on a distributed hypertable compared to a regular
+hypertable. 
 
-### Changing the Number of Data Nodes for a Distributed Hypertable [](changing-data-nodes)
+The planner can consider two types of push downs:
 
-By default, distributed hypertables will use all available data nodes when
-created. But, if you add new data nodes to the distributed database, they are
-not automatically added to existing distributed hypertables, and you need to
-explicitly attach them using [`attach_data_node`][attach_data_node].
+1. **PARTIAL**: a partial push down offloads most of the computation to
+   data nodes, yielding a partial result that is sent back to the
+   access node and finalized by combining all the partials from each
+   data node. An example is computing the `max` temperature; each data
+   node computes a local max and then the access node finalizes the
+   result by computing the `max` over all the data nodes' `max`
+   results:
+   ```
+   SELECT max(temperature) FROM conditions;
+   ```
+2. **FULL**: a full push down offloads all the computation to data
+   nodes and the access node only appends the individual results. To
+   enable full push down, the `GROUP BY` clause must include *all* the
+   partition keys (dimension columns) or *only* the first space
+   partition key. An example is computing the `max` temperature for
+   each location (room1, room2, etc.):
+   
+   ```
+   SELECT location, max(temperature) FROM conditions 
+   GROUP BY location; 
+   ``` 
+   
+   Given that we use `location` as a space partition, each data node
+   can compute the max on its own distinct subset of the data.
+
+
+>:TIP: For push down to work, the settings variable
+>[`enable_partitionwise_aggregate`][partitionwise_agg_guc] must be set
+>to `TRUE`.
+
+#### Limitations of Query Push Down
+
+The query planner might not always be able to push down queries,
+however, or it can only push down parts of it:
+
+- If the query involves a region that saw a change in the partitioning
+  configuration (e.g., the number of space partitions increased), then
+  a full push down might not be possible when it otherwise would
+  be. In our example distributed hypertable, the change in
+  partitioning might lead to data for the same location existing in
+  two chunks that are stored on different data nodes. The planner must
+  therefore avoid *full* push down whenever the query covers the
+  repartitioning boundary. A *partial* push down is still possible,
+  however.
+- If the query includes non-immutable functions and expressions, the
+  planner cannot push down those parts of the query since they cannot
+  be guaranteed to generate a consistent result across each data
+  node. For instance, the [`random()`][random-func] function is
+  [`VOLATILE`][volatility] since it depends on the current seed and
+  the state of the pseudo-random sequence. If the function would be
```

Comment:
```suggestion
  the state of the pseudo-random sequence. If the function were to be
```

Reply:
shouldn't it be "If the function _was_ to be.."?